lib/time.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { getLanguage } from './locale.ts'
+
+export interface FormatDateOptions {
+	/**
+	 * If set then instead of showing seconds since the timestamp show the passed message.
+	 * @default false
+	 */
+	ignoreSeconds?: string | false
+
+	/**
+	 * The relative time formatting option to use
+	 * @default 'long
+	 */
+	relativeTime?: 'long' | 'short' | 'narrow'
+
+	/**
+	 * Language to use
+	 * @default 'current language'
+	 */
+	language?: string
+}
+
+/**
+ * Format a given time as "relative time" also called "humanizing".
+ *
+ * @param timestamp Timestamp or Date object
+ * @param opts Options for the formatting
+ */
+export function formatRelativeTime(
+	timestamp: Date|number = Date.now(),
+	opts: FormatDateOptions = {},
+): string {
+	const options: Required<FormatDateOptions> = {
+		ignoreSeconds: false,
+		language: getLanguage(),
+		relativeTime: 'long' as const,
```

Comment:
What `as const` does here? Isn't string always a const as it is immutable?

Reply:
Type is `'long'` rather than `string`

Reply:
But you have explicit `: Required<FormatDateOptions>` type, which defines `relativeTime` as `'long' | 'short' | 'narrow'` already